README.md

```diff
@@ -1,2 +1,37 @@
 # local_app_prepper
-creates an `input.json` for the LocalApp run
+
+creates `inputs.json` files to be used with the LocalApp
+
+## Introduction :speech_balloon:
+
+Running the LocalApp requires the specification of a least one `inputs.json` file containing the configuration as well as input and output options. To make running the LocalApp more efficient, the `local_app_prepper` will generate several `inputs.json` files:
+
+`inputs.json` | description
+--- | ---
+`demultiplex.json` | running the initial steps to demultiplex the sequencing data
+`tso500_<sample id>.json` | running the DNA/RNA workflow of the TSO500 pipeline from `.fastq` files
+`gather.json` | running the Gather workflow to combine all results into a single directory
+
+## Dependencies :briefcase:
+
+[![beautifulsoup4](https://img.shields.io/badge/beautifulsoup4-4.12.2-blue?color=417fb1)](https://pypi.org/project/beautifulsoup4/)
+
+[![click](https://img.shields.io/badge/click-8.1.7-blue?color=417fb1)](https://pypi.org/project/click/)
+
+[![lxml](https://img.shields.io/badge/lxml-4.9.3-blue?color=417fb1)](https://pypi.org/project/lxml/)
+
+[![python](https://img.shields.io/badge/python-3.11.4-blue?color=417fb1)](https://www.python.org/)
+
+## Usage :rocket:
+
```

Comment:
Suggestion of a bit more expanded usage section follows, please check whether the docker commands make sense. 

Regarding `--input` parameter:
    1. the name `<run id>` does not match definition of the parameter in the script (full folder path) so maybe should be modified to something like `<run sequences folder>` or similar. 
    2. doesn't the folder need to be mounted to the container? If so does it have a specific mounting point?

The suggested readme text follows from here.


There is an image available at `dockerhub` at `inpred/local_app_prepper:latest`. The recommended way of running this tool is via the image. 

### Local Installation

Help text is printed like so:

```bash
$ local_app_prepper.py --help
```

And to produce `inputs.json` files, simply run:

```bash
$ local_app_prepper.py -i <run id> -s <sample 1 id>,<sample 2 id>,...
```

### Docker

Download the image:

```bash
docker pull inpred/local_app_prepper:latest
```

Print help text:

```bash
sudo docker run \
    --rm -it \
    inpred/local_app_prepper:latest \
    python3 local_app_prepper.py --help
```

Run the tool:

```bash
sudo docker run \
    --rm -it \
    inpred/local_app_prepper:latest \
    python3 local_app_prepper.py -i <run_id> -s <sample_1_id>,<sample_2_id>,...
```

### Singularity

Download the image:

```bash
singularity pull local_app_prepper.sif docker://inpred/local_app_prepper:latest
```

Print help text:

```bash
singularity exec \
    --no-home \
    local_app_prepper.sif \
    python3 local_app_prepper.py --help
```

Run the tool:

```bash
singularity exec \
    --no-home \
    local_app_prepper.sif \
    python3 local_app_prepper.py -i <run_id> -s <sample_1_id>,<sample_2_id>,...
```

Reply:
1. yes, you are right, the `<run id>` is indeed the path to the folder. Good catch! 👍 
2. yes, the folder needs to be mounted into the container. It does not have a specific mount path to avoid problems when different paths are chosen. As this tool works as part of the nextflow workflow, nextflow will mount the paths correctly in order to give the tool access to the folder.

Reply:
will add your suggestions for the `README.md` directly to my code base and push the changes.

Reply:
I shortened and summarized what you wrote a tad. I would omit the pulling command because both tools do the pulling as part of starting a container. Also `sudo` is not always required when running docker. `--no-home` seems specific to someone's setup, might not be relevant for everyone; I removed that as well. Also, I assume people start the container and just run the command inside instead of specifying it upon starting the container. I hope it makes sense ☺️

Reply:
Just a few comments from me:
- we currently do require sudo whenever running Docker on our TSO500 analysis server, though it might be possible to change that;
- in my limited experience, the "--no-home" flag was necessary with some Singularity containers (otherwise, some commands/variables would refer to the host OS values rather than container OS values); I am not an expert on this however - there might perhaps be an alternative solution that doesn't require the "--no-home" flag;
- when it comes to pipelines/automated setups, having to enter a container in order to run a command isn't convenient; even if this step is always done manually, having a single command to execute is easier than having to enter an interactive image environment first (also, within TSD, I don't think that one can use the images interactively at all - please correct me here if I am wrong, Tina).

Reply:
- Regarding the `pull` commands: I am not sure what starting process are you referring to,... when I want to execute a command in a container on an offline system, I can do it by pulling the container and then executing the command. By keeping the pull commands there, I think you make people more aware of which tasks are taking place instead of using some very high level commands (which I am not sure are available in TSD) as blackboxes. So I would prefer to keep the commands for this reason.
- Regarding the `--no-home` yes, it is specific setup and usually needed in TSD, in my experience `--cleanenv` usually works fine as well but one of the options is often needed. Saying that, this may be more relevant for how the workflow uses the containers rather than for the documentation about this particular script. I understand that you want to keep this documentation as "clean" as possible by removing unnecessary options, but then maybe small notes could be added about specifics of the local solutions used in InPreD so that people do not spend ages re-inventing something that someone else in InPreD knows very well for ages.
- This is probably rather clear now, but I never run containers interactively in TSD and usually only for debugging purposes outside of TSD. ;) Yeah, I know, feel free to call me a dinosaur. :D In principle, you can run singularity images interactively in TSD, if singularity/aptainer is installed on the access node and you use that node only without accessing the HPC cluster via the submit node. For example, this is how we run fusviz in our TSD project - so I take it that we should be able to test small scripts and their functionality the same way. However. when it comes to running the workflow and its components on the cluster, a different version of singularity is installed there and one is not able to run the containers interactively there, just to execute a command in a container.

Reply:
Hey guys, thanks for your input. I think we are taking about two different things here:
- I was trying to describe how anyone can locally (on your own laptop for example or a linux box) start up the container and use it to "test it out". The tool could be potentially used independently of the workflow, though, I don't think it makes a lot of sense. That is why the documentation should be more on the general side. Individual setups can differ a lot. Just an example: you need `sudo` to run docker, on our linux box the users running docker are in a special group allowing them to execute docker without the `sudo` command. General documentation is supposed to be simple so that anyone could use it and adapt it for their specific setup.
- You are talking about a special case (TSD) which is where you usually work. It makes sense that you need to add certain things in order for the tool to run if you would run it directly. However, it is part of the nextflow workflow and, thus, you won't be executing any of this at the end of the day. Nextflow will basically take the path to your image and execute it via docker or singularity and run the command specified in the workflow script. I think that the nextflow configuration for TSD is relevant because that is what you will be using but local_app_prepper is probably nothing you will run directly in TSD anyways.
- I understand that you need to consider certain things regarding your setup (`docker pull` before you run stuff, `--no-home` to avoid false behavior) but these things are very individual and not everyone needs to run or set them.
- about the `docker run` and `docker pull` thing you can read about it [here](https://epcced.github.io/2020-12-08-Containers-Online/03-running-containers/index.html):
> We could have skipped the docker pull step; if you use the docker run command and you don’t already have a copy of the Docker image, Docker will automatically pull the image first and then run it.

I hope this clarifies a bit of why I chose to write the docs this way.

Reply:
For the first three points, I totally agree. 

Regarding the  `docker pull` and `docker run` - yes, there are many ways to do the same thing, so whatever works.

Reply:
thanks for your quick answer. I was wondering if you had time to review the code (which you maybe already did?). If so and everything is fine, could you please approve the MR? Thank you ☺️